griptape/drivers/audio_transcription/base_audio_transcription_driver.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Optional
+
+from attr import define, field
+
+from griptape.artifacts import TextArtifact, AudioArtifact
+from griptape.events import StartAudioTranscriptionEvent, FinishAudioTranscriptionEvent
+from griptape.mixins import ExponentialBackoffMixin, SerializableMixin
+
+if TYPE_CHECKING:
+    from griptape.structures import Structure
+
+
+@define
+class BaseAudioTranscriptionDriver(SerializableMixin, ExponentialBackoffMixin, ABC):
```

Comment:
Should we name these Drivers `BaseSpeechToTextDriver` for consistency with the inverse Drivers?

Reply:
I'm on the fence about this one. I think I steered a little too far when setting the pattern of drivers based on their artifact interfaces and perhaps too far from describing _what they actually do_ (text in audio out, image in text out, text+image in image out, etc). I think we should probably group drivers by function rather than form in these cases for specificity.

Reply:
I think I've steered a bit too far in the direction of naming drivers based on their artifact interfaces. I think the specificity of this name is helpful, what do you think about adding a similarly specific name to the `BaseTextToSpeechDriver`? `BaseSpeechGenerationDriver`?

Reply:
Yeah thinking about it more I think this current convention is the more "correct" one.

Down to do a rename of `BaseTextToSpeechDriver` (though maybe in a separate PR), what do you think about `BaseAudioGenerationDriver`? This may extend beyond speech in the future?